src/main/java/pl/umk/mat/git2befit/service/user/UserService.java

```diff
@@ -265,11 +265,11 @@ private void sendEmailWithVerificationToken(String email, String token) throws E
         return ResponseEntity.ok().header("Authorization", TOKEN_PREFIX + JWTGenerator.generate(userByEmail.get().getEmail())).build();
     }
 
-    public ResponseEntity<?> sendAgainActivationToken(User user) {
-        Optional<User> savedUser = userRepository.findByEmail(user.getEmail());
+    public ResponseEntity<?> sendAgainActivationToken(String email) {
+        Optional<User> savedUser = userRepository.findByEmail(email);
         if(savedUser.isPresent() && !savedUser.get().isEnable()){
             try {
-                sendEmailWithVerificationToken(user.getEmail(), JWTGenerator.generate(user.getEmail()));
+                sendEmailWithVerificationToken(email, JWTGenerator.generate(email));
```

Comment:
Do tworzenia tokenów aktywacyjnych jest oddzielna metoda statyczna w klasie JWTGenerator. Przyjmuje ona Id użytkownika, którego chcemy aktywować. Jest to wymagane, ponieważ w metodzie activateUser wyciągam z tokena jego id. Przekazanie innego tokena powoduje problemy.
```suggestion
                sendEmailWithVerificationToken(email, JWTGenerator.generateVerificationToken(savedUser.get().getId()));
```
PS już raz to zamieniałem ale w sumie nie wiem czemu tak jest dalej xD